web_external/util/annotationGeometryParser.js

```diff
@@ -1,154 +1,261 @@
+//=============================================================================
+class AnnotationGeometryTrack {
```

Comment:
Overall, this is what I think is unnecessary to refactor, I think a flat design is easier to understand.

Reply:
I have to explicitly disagree. The flat design requires that every place that creates a track (there are at least two) needs to update multiple members. Refactoring moves these operations to the track constructor, which is a much more object-oriented design.

Reply:
I understand your point. I think I need to see when the code in action.

Reply:
Compare:

```javascript
    this._trackIds.add(trackId);
    this._trackRanges.set(trackId, [0, 0]);
    this._enableState.set(trackId, true);
```

...versus:

```javascript
    this._trackMap.set(trackId, new AnnotationGeometryTrack());
```

...and similar for removing a track.

Lookup-wise is a bit more of a wash, but I don't think post-refactor is any *less* logical. Instead of having a map per possible attribute of a track, i.e. `this._someTrackProp.get(id)`, you look up the property *on the track*, i.e. `this._tracks.get(id).someProp`. (To *me*,  this feels more natural, but YMMV.)